libs/offline-tasks/src/main/java/org/opensearch/offline_tasks/client/TaskClient.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.offline_tasks.client;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.offline_tasks.task.Task;
+import org.opensearch.offline_tasks.task.TaskId;
+import org.opensearch.offline_tasks.task.TaskStatus;
+
+import java.util.List;
+
+/**
+ * Client used to interact with Task Store/Queue
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public interface TaskClient {
+
+    /**
+     * Submit a new task to TaskStore/Queue
+     *
+     * @param task Task to be submitted for execution on offline nodes
+     */
+    void submitTask(Task task);
+
+    /**
+     * Claim task from TaskStore/Queue. This ensures no 2 Offline Nodes work on the same task.
+     *
+     * @param taskId TaskId of the task to be claimed
+     * @return true if task is claimed successfully, false otherwise
+     */
+    boolean claimTask(TaskId taskId);
```

Comment:
nit: Thinking if having `submitTask` and `claimTask` in the same interface would be confusing? As I see it, we have `TaskProducer`, `TaskConsumer` and `TaskManager`. But again, having one interface keeps everything at one place.

Reply:
This is a good point, let me think on this. the initial idea is that TaskProducer, TaskConsumer and TaskManager has access to all the interfaces even though they might not use them. But separating them out will add some complexity which might not benefit us much at this point.

If we look at the security angle it might make sense to keep them separate